course.py

```diff
@@ -262,16 +266,26 @@ def add_duplicate_course():
     return redirect(url_for('course.course_info', course_id=course_id, year=year))
 
 
-@course_bp.route('/evaluations/<int:user_id>/<int:current_year>')
+@course_bp.route('/evaluations/<int:user_id>/<int:current_year>', defaults={'evaluation_id': None})
+@course_bp.route('/evaluation/<int:evaluation_id>', defaults={'user_id': None, 'current_year': None})
 @login_required
-def evaluations(user_id, current_year):
-    courses = db.session.query(Course).filter_by(year=current_year).all()
+def evaluations(user_id=None, current_year=None, evaluation_id=None):
+    evaluation = db.session.query(Evaluation).get(evaluation_id) \
+        if evaluation_id else None
+    year = evaluation.course_year if evaluation else current_year
+    courses = db.session.query(Course).filter_by(year=year).all()
+    if evaluation:
+        return render_template('evaluations.html', evaluation=evaluation, current_year=evaluation.course_year,
+                               courses=courses)
 
-    return render_template('evaluations.html', courses=courses, current_year=current_year, user_id=user_id)
+    if user_id and current_year:
+        return render_template('evaluations.html', courses=courses, current_year=current_year, user_id=user_id,
+                               evaluation=evaluation)
```

Comment:
Do not try premature optimization here. Divide these two route handlers. There are already too much if statements in there.